server-common/src/main/java/org/apache/gravitino/server/authentication/OAuthTokenValidatorFactory.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.gravitino.server.authentication;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.gravitino.Config;
+
+/**
+ * Factory for creating appropriate OAuth token validators based on configuration.
+ *
+ * <p>Supports two validation approaches:
+ *
+ * <ul>
+ *   <li><strong>Shared Key Validation</strong> - For providers like Keycloak that use symmetric
+ *       keys
+ *   <li><strong>JWKS Validation</strong> - For providers like Azure, Google, Auth0 that expose JWKS
+ *       endpoints
+ * </ul>
+ */
+public class OAuthTokenValidatorFactory {
+  /**
+   * Create and initialize a token validator based on configuration.
+   *
+   * @param config The configuration object
+   * @return An initialized OAuthTokenValidator
+   * @throws IllegalArgumentException if configuration is invalid
+   */
+  public static OAuthTokenValidator createValidator(Config config) {
+    if (config == null) {
+      throw new IllegalArgumentException("Configuration cannot be null");
+    }
+
+    OAuthTokenValidator validator;
+    if (StringUtils.isNotBlank(config.get(OAuthConfig.JWKS_URI))) {
```

Comment:
Maybe is it suitable to provide a configuration like `gravitino.authenticator.oauth.tokenValidatorClass`?
Another point is what difference between tokenValidator and OAuthProvider? Is `tokenValidator` enough?

Reply:
In the last PR, we have defined ` ProviderType`. Every ProviderType should have its `OAuthProvider`.  What's the relationship between `ProviderType` and `TokenValidator`? How do we use `ProviderType`?

Reply:
So, ProviderType I am planning to use only for the Web UI.
I realize that for the token validation, we don't need provider specific logic.
For web UI we can use it to give provider specific login interfaces (like MSAL for Azure Login). But backend logic can for now be either defaultSignKey or JWKS